Agreement organisation

trevorcummingsCCS opened the pull request:
I have created the extra tables for the Organisations in the Agreement Service. I have also created minimal test data so that the api calls will work.

trevorcummingsCCS committed:
Add tables for suppliers and associated enties

Have added tables to allow the pass back of supplier org details.

trevorcummingsCCS committed:
FKS & Indexes

Added fks and indexes for new tables.

trevorcummingsCCS committed:
Update drop_tables.sql

Added drop commands for the new tables just added.

trevorcummingsCCS committed:
Create create_organisations.sql

trevorcummingsCCS committed:
Update create_organisations.sql

fixed errors

trevorcummingsCCS committed:
missing opening comment

trevorcummingsCCS committed:
Updated tdue to conclave designchnage

trevorcummingsCCS committed:
Updated due to changes to conclave

trevorcummingsCCS committed:
missed one column

trevorcummingsCCS committed:
changed top_level_org_id to ultimate_org_id

trevorcummingsCCS committed:
Update ddl.sql

trevorcummingsCCS committed:
changed organisation_name to legal_name

trevorcummingsCCS committed:
Update create_organisations.sql

trevorcummingsCCS committed:
Create create_lot_suppliers.sql

trevorcummingsCCS committed:
changed comment

trevorcummingsCCS committed:
changed comment block

trevorcummingsCCS committed:
Add contact model

To enable the contact data the agreement service needs to maintain the contact tables in the agreement service

trevorcummingsCCS committed:
Add additional tables from contact model

trevorcummingsCCS committed:
Required to return data for an organisation

trevorcummingsCCS committed:
increase siz of trading_organisation_types.trading_organisation_type_name

trevorcummingsCCS committed:
created data for trading_organisations

trevorcummingsCCS committed:
Create create_contact_method_types.sql

Added data

trevorcummingsCCS committed:
Update create_contact_method_types.sql

trevorcummingsCCS committed:
Create create_contact_point_reasons.sql

trevorcummingsCCS committed:
Update ddl.sql

alter column contact_point_reason from varchar(20) to varchar(100)

trevorcummingsCCS committed:
Update create_trading_organisation_types.sql

trevorcummingsCCS committed:
Create create_contact_methods.sql

trevorcummingsCCS committed:
Create create_contact_points.sql

trevorcummingsCCS committed:
Update ddl.sql

trevorcummingsCCS committed:
Update ddl.sql

trevorcummingsCCS committed:
Update create_lot_suppliers.sql

add additional column to link to a responsible person

trevorcummingsCCS committed:
Create create_people.sql

trevorcummingsCCS committed:
Update create_contact_points.sql

Add party table name so that multiple party tables can be used.

trevorcummingsCCS committed:
Update drop_tables.sql

added people

trevorcummingsCCS committed:
Update ddl.sql

trevorcummingsCCS committed:
Update ddl.sql

trevorcummingsCCS committed:
Update ddl.sql

trevorcummingsCCS committed:
Update ddl.sql

chnaged CONTACT_POINTS to contact_points

trevorcummingsCCS committed:
Update create_people.sql

trevorcummingsCCS committed:
Update ddl.sql

trevorcummingsCCS committed:
Update ddl.sql

trevorcummingsCCS committed:
Update create_people.sql

missing comma

trevorcummingsCCS committed:
Update drop_tables.sql

drop people before organisations

trevorcummingsCCS committed:
Update drop_tables.sql

removed trading_organisation_types as not required

trevorcummingsCCS committed:
Update ddl.sql

removed trading_organisation_types

trevorcummingsCCS committed:
Delete create_trading_organisation_types.sql

Table is not required therefore data is not required.

trevorcummingsCCS committed:
Update drop_tables.sql

moved trading_organisations before organisations

trevorcummingsCCS committed:
added rolea

Added roles for lots and commercial agreements

trevorcummingsCCS committed:
Update ddl.sql

trevorcummingsCCS committed:
Update ddl.sql

trevorcummingsCCS committed:
Update ddl.sql

trevorcummingsCCS committed:
Update drop_tables.sql

added role tables

trevorcummingsCCS committed:
Update ddl.sql

trevorcummingsCCS committed:
Update ddl.sql

add fk

trevorcummingsCCS committed:
Update drop_tables.sql

trevorcummingsCCS committed:
Update ddl.sql

shortened long fk names

trevorcummingsCCS committed:
Update drop_tables.sql

trevorcummingsCCS committed:
Update create_lot_suppliers.sql

trevorcummingsCCS committed:
Update ddl.sql

commit all table name changes

trevorcummingsCCS committed:
Update ddl.sql

trevorcummingsCCS committed:
Update ddl.sql

trevorcummingsCCS committed:
Update ddl.sql

trevorcummingsCCS committed:
Update ddl.sql

amended fks

trevorcummingsCCS committed:
Update drop_tables.sql

added new tables

trevorcummingsCCS committed:
Update ddl.sql

trevorcummingsCCS committed:
Update ddl.sql

trevorcummingsCCS committed:
Update ddl.sql

trevorcummingsCCS committed:
Update ddl.sql

trevorcummingsCCS committed:
Update ddl.sql

trevorcummingsCCS committed:
Update drop_tables.sql

trevorcummingsCCS committed:
Update drop_tables.sql

trevorcummingsCCS committed:
Create create_organisation_role_types.sql

data for lot_organisation_role_types

trevorcummingsCCS committed:
Update create_lot_suppliers.sql

fixed for new columns

trevorcummingsCCS committed:
Create create_commercial_agreement_role_types.sql

trevorcummingsCCS committed:
Delete create_commercial_agreement_suppliers.sql

Table name has changed so file name will change tro reflect this.

trevorcummingsCCS committed:
Create create_commercial_agreement_organisation_roles.sql

add data for commercial_agreement_organisations_roles

trevorcummingsCCS committed:
Create create_lot_people_role_types.sql

add data lot_people_role_types

trevorcummingsCCS committed:
Create create_lot_organisations.sql

replaces the old file lot_suppliers

trevorcummingsCCS committed:
Delete create_lot_suppliers.sql

trevorcummingsCCS committed:
Create create_lot_people_roles.sql

add data for lot_people_roles

trevorcummingsCCS committed:
Update ddl.sql

removed column organisation_id from lot_peole_roles.

trevorcummingsCCS committed:
Update ddl.sql

amended primasry key on lot_people_roles

trevorcummingsCCS committed:
Update create_contact_point_reasons.sql

add a new reason

trevorcummingsCCS committed:
Update create_contact_points.sql

added another record for commercial agreememnt contacts

adrianmilne commented:
The `ddl.sql` version needs increasing (v0.0.2?). I think any changes after this PR we should probably move to versioned ALTER scripts - like we have started in the guided-match folder if everyone agrees

trevorcummingsCCS commented:
Ok will update to version v0.0.2. Might be a good idea just to have a quick five minute chat the three of us regarding these scripts so we are aligned.

trevorcummingsCCS committed:
Update ddl.sql

incremented version number to v.0.0.2

adrianmilne commented on agreements/create_trading_organisations.sql:
```diff
@@ -0,0 +1,9 @@
+Description This file is a script to create data for trading_organisations. This data eventually will reside in Conclave     
+*/
+INSERT INTO public.trading_organisations(
+	trading_organisation_id, trading_organisation_type_id, organisation_id, trading_organisation_name)
```
`trading_organisation_type_id` does not exist on `trading_organisation`

trevorcummingsCCS replied:
Hi @adrianmilne , I just looked at the code in github and trading organisations doesn't have trading_organisation_type_id anymore. Have you got an old version of the code or am I looking in a different branch.

adrianmilne replied:
@trevorcummingsCCS - my comment was from June last year - so probably safe to say an old version of the code 😄

trevorcummingsCCS commented:
Will fix apologies.

trevorcummingsCCS committed:
Update create_trading_organisations.sql

removed reference to deleted column

trevorcummingsCCS committed:
Create create_lot_organisation_roles.sql

File was incorrectly named

trevorcummingsCCS committed:
Delete create_lot_organisations.sql

File has been renamed to lot_organisation_roles.

trevorcummingsCCS committed:
Update create_contact_point_reasons.sql

trevorcummingsCCS committed:
Update ddl.sql

trevorcummingsCCS committed:
Update ddl.sql

trevorcummingsCCS committed:
Update drop_tables.sql

trevorcummingsCCS committed:
Update ddl.sql

trevorcummingsCCS committed:
Update drop_tables.sql

trevorcummingsCCS committed:
Update ddl.sql

trevorcummingsCCS committed:
Delete create_commercial_agreement_contacts.sql

trevorcummingsCCS committed:
Create contact_details.sql

Changed table name from contact_methods to contact_details so have deleted old file and created a new one.

trevorcummingsCCS committed:
Delete contact_details.sql

File named incorrectly

trevorcummingsCCS committed:
Create create_contact_details.sql

Removed the old file contact_details as I should have called it create_contact_details.

trevorcummingsCCS committed:
Delete create_contact_methods.sql

File has been replaced by create_contact_details.sql

trevorcummingsCCS committed:
Update create_contact_points.sql

changed contact_method_id to contact_detail_id

trevorcummingsCCS committed:
Update create_organisations.sql

Removed status as it didn't match conclave

trevorcummingsCCS committed:
Update ddl.sql

removed value_date from lot_attributes and added attribute_uom

trevorcummingsCCS committed:
Update ddl.sql

Added additional organisation columns and the two missing tables.

trevorcummingsCCS committed:
Update drop_tables.sql

Added additional statements for two new tables

trevorcummingsCCS committed:
Update ddl.sql

fixed syntax errors

trevorcummingsCCS committed:
Update ddl.sql

fixed syntax errors

trevorcummingsCCS committed:
Update create_lot_rule_attributes.sql

Amendedscript for additional column attribute_uom

trevorcummingsCCS committed:
Update ddl.sql

Missed status column from organisation

tommyb82 commented:
Hi @trevorcummingsCCS - could you update the `provision-agreements-database.sh` script to include the new `create-*` files - I could try and work out where they need to slot in but probably quicker for you to do it! Thanks 👍

trevorcummingsCCS committed:
Added additional files in.

tommyb82 commented on agreements/provision-agreements-database.sh:
```diff
@@ -19,13 +19,25 @@ export USERNAME=sysadmin
+psql -h $SERVER -d $DATABASE -p $PORT -U $USERNAME -a -q -f create_contact_points.sql
+psql -h $SERVER -d $DATABASE -p $PORT -U $USERNAME -a -q -f create_lot_people_role_types.sql
 psql -h $SERVER -d $DATABASE -p $PORT -U $USERNAME -a -q -f create_lots.sql
+psql -h $SERVER -d $DATABASE -p $PORT -U $USERNAME -a -q -f create_lot_organisation_role_types.sql
```
This file does not exist anymore - produces an error

tommyb82 commented on agreements/drop_tables.sql:
```diff
@@ -9,24 +9,55 @@ Description This file is a script to drop the tables for the Agreements database
+
+DROP TABLE IF EXISTS contact_method_types;
+
+DROP TABLE IF EXISTS commercial_agreement_contacts;
```
This table does not exist anymore

tommyb82 commented on agreements/create_contact_details.sql:
```diff
@@ -0,0 +1,11 @@
+	contact_detail_id, contact_method_type_id, effective_from, effective_to, 
+	virtual_address_value, street_address, locality, region, postal_code, country_code, uprn)
+	VALUES (1, 1, '01-Jan-2016', null, 'test_bat@crowncommercial.gov.uk', null, null, null, null, null, null);
+© 2021 GitHub, Inc.
```
Remove - causing error

tommyb82 commented on agreements/provision-agreements-database.sh:
```diff
@@ -19,13 +19,25 @@ export USERNAME=sysadmin
+psql -h $SERVER -d $DATABASE -p $PORT -U $USERNAME -a -q -f create_lot_organisation_role_types.sql
+psql -h $SERVER -d $DATABASE -p $PORT -U $USERNAME -a -q -f create_organisations.sql
+psql -h $SERVER -d $DATABASE -p $PORT -U $USERNAME -a -q -f create_people.sql
+psql -h $SERVER -d $DATABASE -p $PORT -U $USERNAME -a -q -f create_routes_to_market.sql
```
Renamed to `create_route_to_market.sql`

tommyb82 commented on agreements/create_lot_organisation_roles.sql:
```diff
@@ -0,0 +1,19 @@
+Author      Trevor Cummings
+Description This file is a script to create data for LOT_ORGANISATIONS.   
+*/
+INSERT INTO LOT_ORGANISATION_ROLES (lot_id, 
```
Produces error:
```
psql:create_lot_organisation_roles.sql:19: ERROR:  insert or update on table "lot_organisation_roles" violates foreign key constraint "lot_organisation_lort_fk"
DETAIL:  Key (role_type_id)=(1) is not present in table "lot_organisation_role_types".
```

tommyb82 reviewed (changes requested):
Few errors running the provisioning script (set `\set ON_ERROR_STOP on` in `~/.psqlrc` to catch them 👍

tommyb82 committed:
Fix agreements provisioning script

trevorcummingsCCS committed:
Added Commercial agreement documents

trevorcummingsCCS committed:
fixed syntax error

tommyb82 committed:
remove script output

adrianmilne reviewed (approved):
👍  Awesome - new merged code applied successfully to SBX1
